packages/rest/src/http/route.ts

```diff
@@ -51,9 +51,16 @@ function addSegmentFragment(program: Program, target: Type, pathFragments: strin
   // Don't add the segment prefix if it is meant to be excluded
   // (empty string means exclude the segment)
   const segment = getSegment(program, target);
-  const separator = getSegmentSeparator(program, target);
+
   if (segment && segment !== "") {
-    pathFragments.push(`${separator ?? "/"}${segment}`);
+    // replace the action sentinel !! to ensure proper replacement
```

Comment:
Doesn't this immediately break anyone who (successfully) use @segmentSeparator on a ResourceAction?

Reply:
Mmm... good point.

Reply:
These decorators are primarily used in the `cadl-azure` repo so we should make sure all the tests and samples pass cleanly there.

Reply:
@daviwil that's already been done. https://github.com/Azure/cadl-azure/pull/2287 
However, due to the weirdness of submodules and other such voodoo, I'm not sure the right sequence. But on my machine, all the cadl-azure and cadl tests pass with these two PRs.

Reply:
But instead of updating the old tests, I should duplicate them to make sure `@segmentSeparator` still works.